### DISCUSSION TITLE
How to configure dotfiles module completions?

### DISCUSSION THREAD

**Discussion by TymekDev** [folke/lazydev.nvim#47] [Q&A]
Hey! I just installed lazydev.nvim and I absolutely love it.

Is the suggested way to get completions for my dotfiles? By dotfiles I mean the stuff that I symlink into `~/.config/nvim/`. In particular, I have `~/.config/nvim/lua/tymek` that I would like to get completions for.

So far I have added `~/.config/nvim/lua/` to library for it to work. It got me thinking if there is a better way, though.

**Comment by folke** [comment]
You don't need to add anything. That's all automatic. Don't add libraries yourself.

**Reply by TymekDev** [reply]
Hmm, now that you mention it... I removed the library and it works. _Almost._ It looks like sometimes I get a full module path, i.e. `config.nvim.lua.tymek.whatever` instead of `tymek.whatever`.

It depends on whether the file I open already has `require("tymek.any_module")` (`require("tymek")` is not enough) or not. If there is no `require`, then I get the verbose version.

Any ideas as to why this might be happening?

<details>
<summary>Additional info</summary>

1. I have a symlink `~/.config/nvim` -> `~/personal/dotfiles/config/nvim`.
2. The config I use:
    ```lua
    {
      "folke/lazydev.nvim",
      ft = "lua",
      opts = {},
    }
    ```
3. The behavior doesn't change based on current directory. It is the same when I open `nvim` from `~/personal/dotfiles` and `~/personal/dotfiles/config/nvim`.
4. File's location doesn't change behavior. It is the same for root of my config, `lua/` and `lua/tymek/` directories.

</details>

Edit: I just noticed that `:LazyDev` reports two different `library` setups. If there is `require` in the file, then I get the second `nvim` entry:
```md
## ~/personal/dotfiles
- **runtime** `~/.local/share/bob/v0.10.0/nvim-macos-arm64/share/nvim/runtime/lua`
- **nvim** `~/personal/dotfiles/config/nvim/lua`
```
